# Bonus: Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt

In [2]:
# "tobs" is "temperature observations"
df = pd.read_csv('data/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [3]:
# Convert the date column format from string to datetime
df['date'] = pd.to_datetime(df['date'])
df.dtypes

station            object
date       datetime64[ns]
prcp              float64
tobs                int64
dtype: object

### Compare June and December data across all years 

In [4]:
from scipy import stats

In [5]:
# Filter data for desired months
june_df = df.loc[df['date'].dt.month == 6,:]
december_df = df.loc[df['date'].dt.month == 12,:]

In [6]:
# Identify the average temperature for June
june_avg_temp = june_df['tobs'].mean()

print(f'The average temperature in June is {round(june_avg_temp,2)}')

The average temperature in June is 74.94


In [7]:
# Identify the average temperature for December
december_avg_temp = december_df['tobs'].mean()

print(f'The average temperature in December is {round(december_avg_temp,2)}')

The average temperature in December is 71.04


In [8]:
# Create collections to run unpaired t-test
june_temps = june_df['tobs'].tolist()
dec_temps = december_df['tobs'].tolist()


In [9]:
# Run unpaired t-test
stats.ttest_ind(june_temps, dec_temps)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

In the case of comparing mean temperatures between June and December, an unpaired t-test was used to determine if the difference in means is statistically significant. The reason why an unpaired t-test was chosen as opposed to a paired is because in this case, the samples of temperature measurements were taken at different points in time, and there was a different number of measurements captured for each month. The 2 sets of measurements can therefore be considered independent groups, thus warranting an unpaired t-test. After running the result, a p-value of 3.9 * 10^-191 was observed (see above cell). This indicates that there is essentially a 0% probability that the differences in means were a result from chance. Thus, we can conclude that the difference in mean temperatures between June and December is in fact statistically significant.